### PR TITLE
SlackBot Praise Review

### PR DESCRIPTION
**Pull request by AndrewHUNGNguyen**
Main files of focus are the following:
- praise-bot/functions/save_praise.ts
- praise-bot/manifest.ts
- praise-bot/triggers/give_kudos.ts
- praise-bot/workflows/give_kudos.ts

The struggle that I am encountering inside of **praise-bot/functions/save_praise.ts** is knowing how to use the client object in the function header at **line 40** to extract the sender User ID to pass in as part of the API request

Here is a sample JSON Body object to send for the slackbot:
{
        "praise_sender": "ABC123456DEF",
        "praise_receiver":"U04PSMJ0GRX",
        "praise_channel":"C07LQ4VD1JA",
        "praise_message":"Mock API request without the gif URL, should append Null to praise_gif field"
}, which is the object where a user DOES NOT choose a vibe in the Praise Interactive Form

OR

{
        "praise_sender": "ABC123456DEF",
        "praise_receiver":"U04PSMJ0GRX",
        "praise_channel":"C07LQ4VD1JA",
        "praise_message":"Mock API request with the gif URL, should leave as is",
        "praise_gif": "http://hello-world.com/"
}, which is the object where a user DOES NOT choose a vibe in the Praise Interactive Form

Below is an attempted High-level system design to see how a user is to interact with the slackbot and how data is to be represented and how the backend files and slackbot typescript files interact with each other.
[Praise Slackbot Drawings.pdf](https://github.com/user-attachments/files/17021098/Praise.Slackbot.Drawings.pdf)

### CODE REVIEW

File: praise-bot/functions/save_praise.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        },                
+      },
+      required: ["doer", "channel", "message"],
+    },
+  });
+
+  export default SlackFunction(SavePraiseFunction, async ({ inputs, client, env }) => {
+    const debugMode = isDebugMode(env);
+    if (debugMode) {
+      console.log(`translate inputs: ${JSON.stringify(inputs)}`);
```
Comment (gregv) [inline praise-bot/functions/save_praise.ts:43]: @AndrewHUNGNguyen - can you send me this complete object on Slack (incase it has sensitive info in it)?

My guess is that inputs.doer will have the Slack ID in there and you can use that for the backend call

Comment (AndrewHUNGNguyen) [reply praise-bot/functions/save_praise.ts:43]: Hi @gregv if you are asking about line 43, here it is

image

File: praise-bot/functions/save_praise.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ * Defines the SavePraiseFunction input and output parameters
+ */
+export const SavePraiseFunction = DefineFunction({
+    callback_id: "save_praise",
+    title: "Save a Praise",
+    description: "Saves data about a praise based on user input choices",
+    source_file: "functions/save_praise.ts", // The file with the exported function handler
+    input_parameters: {
+      properties: {
+        sender: {
```
Comment (gregv) [inline praise-bot/functions/save_praise.ts:15]: @AndrewHUNGNguyen looks like this then is the variable you want

File: praise-bot/functions/save_praise.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        message: {
+            type: Schema.types.string,
+            description: "The message content of the praise",
+        },
+        vibe: {
+            type: Schema.types.string,
+            description: "The energy of the praise",
+        },                
+      },
+      required: ["doer", "channel", "message"],
```
Comment (gregv) [inline praise-bot/functions/save_praise.ts:36]: Is "sender" not also required here @AndrewHUNGNguyen ?

Comment (AndrewHUNGNguyen) [reply praise-bot/functions/save_praise.ts:36]: I was debating whether or not to make it as a required variable.

File: praise-bot/functions/save_praise.ts (hunk 2 above)
Comment (AndrewHUNGNguyen) [reply praise-bot/functions/save_praise.ts:15]: Yes the sender variable is what I want to send as data about the User that triggered the praise slackbot

File: praise-bot/.devcontainer/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+RUN apt-get install -y unzip
+
+# Run the container as a user
+RUN useradd -m -s /bin/bash slackdev
+RUN chown slackdev /usr/local/bin
+USER slackdev
+
+# Install the project runtime
+RUN curl -fsSL https://deno.land/install.sh | sh
+RUN export DENO_INSTALL="/home/slackdev/.deno"
```
Comment (gregv) [inline praise-bot/.devcontainer/Dockerfile:17]: What's this for @AndrewHUNGNguyen ? You normally don't want hardcoded directories in Dockerfiles

**Review by gregv — APPROVED** [review]

File: praise-bot/.devcontainer/Dockerfile (hunk 4 above)
Comment (AndrewHUNGNguyen) [reply praise-bot/.devcontainer/Dockerfile:17]: Hi Greg, I think this is an auto-generated file when I copied the starter code.

**Pull request merged**